feature-account-api/src/main/java/io/novafoundation/nova/feature_account_api/data/repository/addAccount/AddAccountRepository.kt

```diff
@@ -9,13 +9,35 @@ interface AddAccountRepository<T> {
 
 sealed interface AddAccountResult {
 
-    val metaId: Long
+    sealed interface HadEffect : AddAccountResult
 
-    val type: LightMetaAccount.Type
+    interface SingleAccountChange {
 
-    class AccountAdded(override val metaId: Long, override val type: LightMetaAccount.Type) : AddAccountResult
+        val metaId: Long
+    }
 
-    class AccountChanged(override val metaId: Long) : AddAccountResult
+    class AccountAdded(override val metaId: Long, val type: LightMetaAccount.Type) : HadEffect, SingleAccountChange
 
-    class NoOp(override val metaId: Long) : AddAccountResult
+    class AccountChanged(override val metaId: Long, val type: LightMetaAccount.Type) : HadEffect, SingleAccountChange
+
+    class Batch(val updates: List<HadEffect>) : HadEffect
+
+    object NoOp : AddAccountResult
```

Comment:
I think it would be better to name it without abbreviation

Reply:
It is quite a standard and well-known term: https://ru.wikipedia.org/wiki/NOP. I think its fine having it like that especially considering that a lot of codebases do the same (e.g. Substrate does)